Tests/Creation/Frame/Content/String/ID3CommentTypesFrameCreatorTest.swift

```diff
@@ -25,13 +31,20 @@ class ID3CommentTypesFrameCreatorTest: XCTestCase {
                 content: "test"
         )
 
-        #warning("this is going to fail because I don't know how to adapt it for the new type")
-        XCTAssertEqual(frameBytes, [0x22, 0x11, 0x00, /* insert .und in bytes*/ /* insert "description in bytes followed by null character*/ 0x01, 0x00, 0x00, 0x74, 0x65, 0x73, 0x74, 0x00, 0x00])
-    }
-}
-
-class MockCommentTypesStringToBytesAdapter: StringToBytesAdapter {
-    func adapt(string: String, for version: ID3Version) -> [UInt8] {
-        return [0x01, 0x00, 0x00, 0x74, 0x65, 0x73, 0x74, 0x00, 0x00]
+        XCTAssertEqual(
+          frameBytes,
+          [
+            0x22, // Identifier (See above.)
+            0x00, 0x00, 0x00, 0x2D, // Size
+            0x01, // UCS‐2
+            0xFF, 0xFE, 0x75, 0x00, 0x6E, 0x00, 0x64, 0x00, // “und”
+            0xFF, 0xFE, 0x64, 0x00, 0x65, 0x00, 0x73, 0x00, 0x63, 0x00, 0x72, 0x00, 0x69, 0x00, 0x70, 0x00, 0x74, 0x00, 0x69, 0x00, 0x6F, 0x00, 0x6E, 0x00, 0x00, 0x00, // “description”
+            0xFF, 0xFE, 0x74, 0x00, 0x65, 0x00, 0x73, 0x00, 0x74, 0x00 // “test”
+          ]
+        )
+        #warning("Why is the size four bytes? The specification says it should be three.")
+        #warning("Encoding flag is UCS‐2, which is not UTF‐16.")
+        #warning("Language code is supposed to be ASCII, not UTF‐16.")
+        #warning("The leading FF FE on each string is a byte order mark. Is that valid?")
```

Comment:
I have no idea on any of that. @chicio might know?

Reply:
Since I wrote that I realized that the different versions have different sizes for some fields and I was only looking at one. That could be why it differs from what I was reading. I’ve been working from version 2 as specified [here](http://id3lib.sourceforge.net/id3/id3v2-00.txt): I haven’t been able to find a specification for versions 3 or 4, since the [official site (id3.org)](http://id3.org) just gives me a bad gateway error, and that is where most sites link to for more information.

Reply:
Ah so the bad gateway error isn't just me. Good to know.

I can help with the documentation, however. The TagLib source code I've been playing around with includes the specifications in text files.

[version 2](https://github.com/NCrusher74/TagLib/blob/master/taglib-src/mpeg/id3v2/id3v2.2.0.txt)
[version 3](https://github.com/NCrusher74/TagLib/blob/master/taglib-src/mpeg/id3v2/id3v2.3.0.txt)
[version 4](https://github.com/NCrusher74/TagLib/blob/master/taglib-src/mpeg/id3v2/id3v2.4.0-frames.txt)
[version 4 structure](https://github.com/NCrusher74/TagLib/blob/master/taglib-src/mpeg/id3v2/id3v2.4.0-structure.txt)
And since I can't find it in the TagLib files, here's the Mutagen copy of the document about [chapter and toc frames](https://mutagen-specs.readthedocs.io/en/latest/id3/id3v2-chapters-1.0.html)